### PR TITLE
RTN-1557: Update Feedback component to receive Caption as a new child component

### PR DESCRIPTION
**Pull request by matheushdsbr**
[JIRA Issue](https://gympass.atlassian.net/browse/)

## Description 📄

In this PR, `Feedback.Caption` has been added as a new child of the Feedback component.

To give context, we need to add information on the Cancellation Success screen for primary users. To do this, we need to update the feedback component with an option to add one more description (Caption). 
This will give the user more information about their cancellation. 

To more details: [task history](https://gympass.atlassian.net/browse/RTN-1539) and [figma](https://www.figma.com/design/RofGB6OCJEP7kcguw3TpQy/%5BRetention%5D-Cancel-and-Pause-(Rebranding)?node-id=13550-88709&node-type=instance&t=X3u0dllQjm13501y-0)

## Platforms 📲

- [x] Web
- [ ] Mobile

## Type of change 🔍

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested? 🧪

[Enter the tips to test this PR]

- [x] Unit Test
- [ ] Snapshot Test

## Checklist: 🔍

- [x] My code follows the contribution guide of this project [Contributing Guide](https://github.com/Gympass/yoga/blob/master/CONTRIBUTING.md)
- [x] Layout matches design prototype: [FIGMA](https://www.figma.com/design/RofGB6OCJEP7kcguw3TpQy/%5BRetention%5D-Cancel-and-Pause-(Rebranding)?node-id=13550-88709&node-type=instance&t=X3u0dllQjm13501y-0)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### Web

| V3Theme  | V2Theme |
| ------ | ----- |
| Screenshot 2024-11-19 at 10 09 46 | Screenshot 2024-11-19 at 08 57 49 |

### Mobile/Responsive

| V3Theme  | V2Theme |
| ------ | ----- |
| Screenshot 2024-11-19 at 10 15 56 | Screenshot 2024-11-19 at 10 17 28 |

### CODE REVIEW

File: packages/yoga/src/Feedback/web/StyledFeedback.jsx [hunk 1]
```diff
@@ -94,6 +94,14 @@ export const SecondaryButton = styled(Button.Text)`
   ${buttonsStyles}
 `;
 
+export const Caption = styled(Text.Body1).attrs(() => ({
+  as: 'p',
+  bold: true,
+}))`
+  color: #6b6b78;
```
Comment (flavia-moraes) [inline packages/yoga/src/Feedback/web/StyledFeedback.jsx:101]: If possible, use the color tokens from our theme

**Review by rodrigo-tavares — APPROVED** [review]

File: packages/yoga/src/Feedback/web/StyledFeedback.jsx [hunk 2]
```diff
@@ -94,6 +94,21 @@ export const SecondaryButton = styled(Button.Text)`
...
+  as: 'p',
+  bold: true,
+}))`
+  ${({
+    theme: {
+      yoga: { colors },
+    },
+  }) =>
+    css`
+      color: ${colors.text.secondary};
```
[the remark this answers is not in the document]
Comment (matheushdsbr) [reply packages/yoga/src/Feedback/web/StyledFeedback.jsx:107]: This change doesn't make sense, the code is aligned

**Review by ericcleao — APPROVED** [review]

File: packages/yoga/src/Feedback/web/Feedback.jsx [hunk 3]
```diff
@@ -75,6 +78,9 @@ function Feedback({ variant, title, description, children, center, ...props }) {
             {description}
           </Text.Body1>
         </TextContainer>
+
+        {captionElement ? <Box mt="medium">{captionElement}</Box> : null}
```
Comment (flavia-moraes) [inline packages/yoga/src/Feedback/web/Feedback.jsx:82]: Shouldn't this content be text?

Comment (matheushdsbr) [reply packages/yoga/src/Feedback/web/Feedback.jsx:82]: This content is a `text`, `captionElement` is a `Text.Body1`

This happens in line 50 where we have:
`if (isChildFromComponent(child, Caption)) captionElement = child;`

**Pull request merged**